test/Microsoft.ML.FSharp.Tests/SmokeTests.fs

```diff
@@ -0,0 +1,132 @@
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Newtonsoft.Json.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/System.CodeDom.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/System.Threading.Tasks.Dataflow.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.CpuMath.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.Data.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.Transforms.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.ResultProcessor.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.PCA.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.KMeansClustering.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.FastTree.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.Api.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.Sweeper.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.StandardLearners.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/Microsoft.ML.PipelineInference.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/xunit.core.dll" 
+#r @"../../bin/AnyCPU.Debug/Microsoft.ML.FSharp.Tests/net461/xunit.assert.dll" 
+#r "System" 
+#r "System.ComponentModel.Composition" 
+#r "System.Core" 
+#r "System.Xml.Linq" 
+
+// Later tests will add data import using F# type providers:
+//#r @"../../packages/fsharp.data/3.0.0-beta4/lib/netstandard2.0/FSharp.Data.dll" // this must be referenced from its package location
+
+let _load  =
+    // See https://github.com/dotnet/machinelearning/issues/401: forces the loading of ML.NET assemblies
+    [ typeof<Microsoft.ML.Runtime.Transforms.TextAnalytics>; 
+      typeof<Microsoft.ML.Runtime.FastTree.FastTree> ]
+
+#endif
+
+//================================================================================
+// The tests proper start here
+
+#if !INTERACTIVE
+namespace Microsoft.ML.FSharp.Tests
+#endif
+
+open System
+open Microsoft.ML
+open Microsoft.ML.Data
+open Microsoft.ML.Transforms
+open Microsoft.ML.Trainers
+open Microsoft.ML.Runtime.Api
+open Xunit
+
+module SmokeTest1 = 
+
+    type SentimentData() =
+        [<Column(ordinal = "0"); DefaultValue>]
+        val mutable SentimentText : string
+        [<Column(ordinal = "1", name = "Label"); DefaultValue>]
+        val mutable Sentiment : float32
+
+    type SentimentPrediction() =
+        [<ColumnName "PredictedLabel"; DefaultValue>]
+        val mutable Sentiment : bool
+
+    [<Fact>]
+    let ``FSharp-Sentiment-Smoke-Test`` () =
+
+        let testDataPath = __SOURCE_DIRECTORY__ + @"/../data/wikipedia-detox-250-line-data.tsv"
```

Comment:
Does `__SOURCE_DIRECTORY__ ` work in non-interactive mode?

Reply:
yes it does